src/commands/project/pipeline.ts

```diff
@@ -66,6 +66,26 @@ export const checkDependencies = (projectPath: string): void => {
   }
 };
 
+export const validateRepoUrl= (
+  opts: CommandOptions,
+  gitOriginUrl: string
+): string => {
+  try {
+    let repoUrl = ''
+    if (!opts.repoUrl) {
```

Comment:
i think this logic could be made simpler by inverting the repoUrl check condition and returning early and only calling `getRepositoryUrl` once:

```
if (opts.repoUrl) {
    return opts.repoUrl;
}

const repoUrl = getRepositoryUrl(gitOriginUrl);

if (!repoUrl) {
    throw Error(`Repo url not defined. Are you in a spk project folder?`);
}

return repoUrl;
```

Reply:
I'm not sure if what you're proposing is the same logic I am trying to have the code achieve. What I am trying to do is check for the the argument value that is passed in for `repoUrl`, if it exist then set variable and proceed, skipping the need to confirm with `gitOriginUrl`. If it doesn't exist then check if it is possible obtain the `repoUrl` using `gitOriginUrl`. If that is not possible then throw an error, and if possible then set the variable appropriately. 

From your code, it seems like we first check to see if the argument was passed in, then we set the variable using `gitOriginUrl`. But we never prioritize the passed in argument to set the variable, and always rely on `gitOriginUrl` to determine the repo url, which I believe is not what we want.

Reply:
@bnookala I tested your snippet of code, and it appears to work! Thanks for the feedback. Marking resolved.